renderer/src/common/components/settings/registry/registry-source-field.tsx

```diff
@@ -12,6 +12,57 @@ import { Input } from '../../ui/input'
 import type { RegistryFormData } from './schema'
 import { FilePickerInput } from '../../ui/file-picker-input'
 
+function renderDescription(type: RegistryFormData['type']) {
```

Comment:
nit: this could be a component like `<DescriptionContent />`


but the bigger thing is, I guess it could make more sense to colocate everything related to a registry server type. In other words, instead of putting parallel branching here, you could avoid extra branching by just making this an attribute on each item in `REGISTRY_SOURCE_CONFIG`:


```tsx

const REGISTRY_SOURCE_CONFIG = {
  local_path: {
    label: 'Registry File Path',
    placeholder: '/path/to/registry.json',
    useFilePicker: true,
    description: <>Description comes here</>
  },
  // ... etc ...
} as const satisfies Record<
  Exclude<RegistryFormData['type'], 'default'>,
  { label: string; placeholder: string; useFilePicker: boolean, description: Rect.ReacNode}
>
```

Reply:
so I guess that way you won't need a component or a function of any kind